src/Microsoft.ML.Data/DataView/DataViewConstructionUtils.cs

```diff
@@ -855,16 +857,20 @@ public MetadataInfo(string kind, T value, ColumnType metadataType = null)
             else
             {
                 // Make sure that the types are compatible with the declared type, including whether it is a vector type.
-                if (isVector != metadataType.IsVector)
+                VectorType metdataVectorType = metadataType as VectorType;
+                bool metdataIsVector = metdataVectorType != null;
```

Comment:
can you spell them `metadataSomething` and not `metdataSomething` I don't see any reason to short it.

Reply:
Oops. Type-o.  Fixed.